cxx/pclean/schema_helper.cc

```diff
@@ -98,6 +195,9 @@ std::vector<std::string> reorder_domains(
 
 T_schema PCleanSchemaHelper::make_hirm_schema() {
   T_schema tschema;
+
+  // For every scalar variable, make a clean relation with the name
```

Comment:
Could you add a check that all scalar variables appear in the query? If we define clean relations that aren't observed and aren't the base relation for a noisy relation that's observed, I'm not sure what happens downstream (and that seems unintended).

Reply:
I think we should support scalar variables not appearing in the query, because I think we should support model specification being an independent thing from query specification.  That is, if you spend a lot of time developing a fancy probabilistic model of some domain, it would be great if you could just hand it to me and I could use it just for the fewer columns that interest me (or that I have available) without having to edit the model down.

It would be great if the IRM/HIRM code could support "totally unobserved" clean relations, because that is the most natural place for it.  (For example, the same issue arises when not using schema generated by pclean).  I'll create an issue so we can discuss this further and figure out a fix.